ios/FluentUI/Card Nudge/CardNudge.swift

```diff
@@ -0,0 +1,265 @@
+                           actionButtonTitle: String? = nil,
+                           actionButtonAction: CardNudgeButtonAction? = nil,
+                           dismissButtonAction: CardNudgeButtonAction? = nil) {
+        self.init(style: style, title: title)
+
+        self.subtitle = subtitle
+        self.mainIcon = mainIcon
+        self.accentText = accentText
+        self.actionButtonTitle = actionButtonTitle
+        self.actionButtonAction = actionButtonAction
+        self.dismissButtonAction = dismissButtonAction
+    }
+}
+
+public struct CardNudge: View {
+    @ObservedObject var tokens: CardNudgeTokens
+    @ObservedObject var state: CardNudgeStateImpl
+
+    @ViewBuilder
+    var icon: some View {
+        if let icon = state.mainIcon {
+            ZStack {
+                Circle()
+                    .frame(width: tokens.circleSize, height: tokens.circleSize)
+                    .foregroundColor(Color(tokens.buttonBackgroundColor))
+                Image(uiImage: icon)
+                    .renderingMode(.template)
+                    .frame(width: tokens.iconSize, height: tokens.iconSize, alignment: .center)
+                    .foregroundColor(Color(tokens.accentColor))
+            }
+            .padding(.trailing, tokens.innerPadding)
+        }
+    }
+
+    private var hasSecondTextRow: Bool {
+        state.accentIcon != nil || state.accentText != nil || state.subtitle != nil
+    }
+
+    @ViewBuilder
+    var textContainer: some View {
+        VStack(alignment: .leading, spacing: tokens.interTextVerticalPadding) {
+            Text(state.title)
+                .font(.subheadline)
+                .fontWeight(.medium)
+                .lineLimit(1)
+                .foregroundColor(Color(tokens.textColor))
+
+            if hasSecondTextRow {
+                HStack(spacing: tokens.accentPadding) {
+                    if let accentIcon = state.accentIcon {
+                        Image(uiImage: accentIcon)
+                            .renderingMode(.template)
+                            .frame(width: tokens.accentIconSize, height: tokens.accentIconSize)
+                            .foregroundColor(Color(tokens.accentColor))
+                    }
+                    if let accent = state.accentText {
+                        Text(accent)
+                            .font(.subheadline)
+                            .layoutPriority(1)
+                            .lineLimit(1)
+                            .foregroundColor(Color(tokens.accentColor))
+                    }
+                    if let subtitle = state.subtitle {
+                        Text(subtitle)
+                            .font(.subheadline)
+                            .lineLimit(1)
+                            .foregroundColor(Color(tokens.subtitleTextColor))
+                    }
+                }
+            }
+        }
+    }
+
+    @ViewBuilder
+    var buttons: some View {
```

Comment:
our existing fluent pill style buttons aint good enough?

Reply:
The existing buttons in this branch are UIKit-based, not SwiftUI. There's probably a way to host one of our existing UIKit buttons here, but I think it's likely more trouble than it's worth.

Once SwiftUI buttons are merged to `main` and usable via a SwiftUI client, then we can replace this.

Reply:
ah right.